vbjax/loops.py

```diff
@@ -13,9 +13,9 @@ def heun_step(x, dfun, dt, *args, add=0, adhoc=None):
     """
     adhoc = adhoc or (lambda x,*args: x)
     d1 = dfun(x, *args)
-    xi = adhoc(x + dt*d1 + add, *args)
+    xi = adhoc(x, dt*d1 + add, *args)
```

Comment:
what's going on here? the `adhoc` function should have access to the value of `xi = x + dt*d1`